subgraphs/burrow/src/handlers/farm.ts

```diff
@@ -68,56 +62,49 @@ export function handleAddAssetFarmReward(event: EventData): void {
   }
 
   const reward_amount = data.get("reward_amount");
-  if (!reward_amount) {
-    log.warning("handleAddAssetFarmReward() :: reward_amount not found", []);
-    return;
-  }
-
   const rewardToken = getOrCreateRewardToken(reward_token, farmType);
 
   const market = getOrCreateMarket(farm);
-  const reward_tokens = market.rewardTokens!;
+  const rewardTokens = market.rewardTokens!;
   // push if reward token does not exists, get index
-  let reward_token_index = reward_tokens.indexOf(rewardToken.id);
+  let reward_token_index = rewardTokens.indexOf(rewardToken.id);
   if (reward_token_index == -1) {
-    reward_tokens.push(rewardToken.id);
-    market.rewardTokens = reward_tokens;
-    reward_token_index = reward_tokens.length - 1;
+    rewardTokens.push(rewardToken.id);
+    market.rewardTokens = rewardTokens;
+    reward_token_index = rewardTokens.length - 1;
```

Comment:
You need to be careful since this array will get sorted alphabetically even if you do not do that: https://github.com/messari/subgraphs/blob/master/docs/ERRORS.md#array-sorting-when-querying

But the emissions array will not, so you may have mismatches

Reply:
Take a look at the docs I linked, tl;dr `market.rewardTokens` automatically gets sorted while `market.rewardTokenEmissionsAmount` doesn't so if you push to the end always rewardTokens might change order and the emissions stays